src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -315,7 +315,14 @@ public IEnumerable<TTransformed> Transform(IEnumerable<TExtracted> items, Docume
         {
             using (var transformer = GetTransformer(context))
             {
-                transformer.Initialize(debugMode: _testMode != null);
+                try
+                {
+                    transformer.Initialize(debugMode: _testMode != null);
+                }
+                catch (JavaScriptParseException e)
+                {
+                    HandleTransformationScriptParseException(stats, e);
```

Comment:
We need to break here and return empty transformation result, don't we? Otherwise we'll attempt to iterate over `items`